clean.py

```diff
@@ -47,6 +53,14 @@ def apply_function(function_name: str, ds: Dataset, num_proc: int, batch_size: i
         logger.info(f"     Removed samples: {len(ds) - len(filtered_ds)} samples")
         logger.info(f"     Removed percentage: {(len(ds) - len(filtered_ds)) / len(ds) * 100:.2f} %")
         return filtered_ds
+    elif function_name in DEDUPS:
+        dedup_function = DEDUPS[function_name]
+        deduplicated_ds = dedup_function(ds, num_proc=num_proc, batch_size=batch_size)
+        logger.info(f"Applied deduplication: {function_name}")
```

Comment:
Added a comment to the other PR as well: can we outsource all the logging and auxiliary tasks to dedicated function to make reading the main function logic a bit easier?

Reply:
Okay makes sense will do that in the next PR if that's alright